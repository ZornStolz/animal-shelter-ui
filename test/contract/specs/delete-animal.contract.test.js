import { provider } from "../config/init-pact";
import { Matchers } from "@pact-foundation/pact";
import { AnimalController } from "../../../controllers";

describe("Animal Service", () => {
  describe("When a request to delete an animal is made", () => {
    beforeAll(async () => {
      await provider.setup();
      await provider.addInteraction({
        uponReceiving: "a request to delete an animal",
        state: "delete animal",
        withRequest: {
          method: "DELETE",
          path: "/animals/manchas",
        },
        willRespondWith: {
          status: 204,
        },
      });
    });

    test("should return the correct data", async () => {
      const response = await AnimalController.delete("manchas");
      expect(response).toMatchSnapshot();
      await provider.verify();
    });

    afterAll(() => provider.finalize());
  });
});
