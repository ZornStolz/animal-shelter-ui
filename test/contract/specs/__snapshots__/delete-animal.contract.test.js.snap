// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Service When a request to delete an animal is made should return the correct data 1`] = `
Object {
  "config": Object {
    "adapter": [Function],
    "baseURL": "http://localhost:8080",
    "data": undefined,
    "headers": Object {
      "Accept": "application/json, text/plain, */*",
    },
    "maxBodyLength": -1,
    "maxContentLength": -1,
    "method": "delete",
    "timeout": 0,
    "transformRequest": Array [
      [Function],
    ],
    "transformResponse": Array [
      [Function],
    ],
    "transitional": Object {
      "clarifyTimeoutError": false,
      "forcedJSONParsing": true,
      "silentJSONParsing": true,
    },
    "url": "animals/manchas",
    "validateStatus": [Function],
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
  },
  "data": "",
  "headers": Object {},
  "request": XMLHttpRequest {},
  "status": 204,
  "statusText": "No Content ",
}
`;
